boards/FSM/Src/Io/Io_AcceleratorPedals.c

```diff
@@ -52,8 +41,30 @@ float Io_AcceleratorPedals_GetAngle(void)
 
     return pedal_travel_angle * 180 / (float)M_PI + 16.3f;
 }
-
 float Io_AcceleratorPedals_GetPapps(void)
 {
-    return Io_AcceleratorPedals_GetAngle() / 30.0f * 100.0f;
+    float primary_angle = Io_AcceleratorPedals_GetPrimaryAngle();
+    if(primary_angle == NAN) return NAN;
+    return Io_AcceleratorPedals_GetPrimaryAngle() / 30.0f * 100.0f;
+}
+
+//TODO Implement the secondary pedal angle IO function
+float Io_AcceleratorPedals_GetSecondaryAngle(void)
+{
+    float pedal_voltage = 0.5f;
+    if(Io_AcceleratorPedals_PedalVoltageAlarm(pedal_voltage)){
+        return NAN;
+    }
+    return (float)M_PI;
+}
```

Comment:
If this function is never used outside outside of the .c file, it should be marked static and removed from the header file. 
Looks like you could just remove it altogether though and put logic in GetSapps?

Reply:
removed.